src/mono/wasm/runtime/types/emscripten.ts

```diff
@@ -52,8 +52,12 @@ declare interface EmscriptenModule {
     removeRunDependency(id: string): void;
     addRunDependency(id: string): void;
 
-    preInit?: (() => Promise<void>)[];
+    ready: Promise<unknown>;
+    preInit?: (() => void | Promise<void>)[];
+    preRun?: (() => void | Promise<void>)[];
+    postRun?: (() => void | Promise<void>)[];
     onRuntimeInitialized?: () => void;
```

Comment:
In `exports.ts` we are assigning a `mono_wasm_on_runtime_initialized` to it, which returns `promise`. 
Is `void` correct type or should it be the `promise`?

Reply:
`void` is correct. In order to delay `run` we have to `addRunDependency/removeRunDependency` before/after async work.
Thanks!